src/test/java/org/folio/search/controller/BrowseTypedCallNumberIrrelevantResultTest.java

```diff
@@ -0,0 +1,173 @@
+import static org.folio.search.support.base.ApiEndpoints.instanceCallNumberBrowsePath;
+import static org.folio.search.utils.TestConstants.TENANT_ID;
+import static org.folio.search.utils.TestUtils.cnBrowseItem;
+import static org.folio.search.utils.TestUtils.getShelfKeyFromCallNumber;
+import static org.folio.search.utils.TestUtils.parseResponse;
+import static org.folio.search.utils.TestUtils.randomId;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.folio.search.domain.dto.CallNumberBrowseResult;
+import org.folio.search.domain.dto.Instance;
+import org.folio.search.domain.dto.Item;
+import org.folio.search.domain.dto.ItemEffectiveCallNumberComponents;
+import org.folio.search.support.base.BaseIntegrationTest;
+import org.folio.search.utils.CallNumberUtils;
+import org.folio.spring.test.type.IntegrationTest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+@IntegrationTest
+class BrowseTypedCallNumberIrrelevantResultTest extends BaseIntegrationTest {
+
+  private static final Instance[] INSTANCES = instances();
+  private static final Map<String, Instance> INSTANCE_MAP =
+    Arrays.stream(INSTANCES).collect(toMap(Instance::getTitle, identity()));
+
+  @BeforeAll
+  static void prepare() {
+    setUpTenant(INSTANCES);
+    enableFeature(BROWSE_CN_INTERMEDIATE_VALUES);
+  }
+
+  @AfterAll
+  static void cleanUp() {
+    removeTenant();
+  }
+
+  @Test
+  void browseByCallNumber_browsingAroundWithEnabledIntermediateValues() {
+    var request = get(instanceCallNumberBrowsePath())
+      .param("callNumberType", "dewey")
+      .param("query", prepareQuery("typedCallNumber >= {value} or typedCallNumber < {value}", "308 H977"))
+      .param("limit", "10")
+      .param("highlightMatch", "true")
+      .param("precedingRecordsCount", "5")
+      .param("expandAll", "true");
+    var actual = parseResponse(doGet(request), CallNumberBrowseResult.class);
+    var expected = new CallNumberBrowseResult().totalRecords(18).prev("3308 H976").next("3308 H981").items(List.of(
+      cnBrowseItem(INSTANCE_MAP.get("instance #01"), "308 H976"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #10"), "308 H970"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #11"), "308 H972"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #11"), "308 H973"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #11"), "308 H974"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #09"), "308 H977", true),
+      cnBrowseItem(INSTANCE_MAP.get("instance #01"), "308 H978"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #10"), "308 H979"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #02"), "308 H980"),
+      cnBrowseItem(INSTANCE_MAP.get("instance #08"), "308 H981")
+    ));
+    expected.setItems(CallNumberUtils.excludeIrrelevantResultItems("dewey", expected.getItems()));
```

Comment:
Why we can't just have `expected` in final form and do `expected.setItems(CallNumberUtils.excludeIrrelevantResultItems("dewey", expected.getItems()))` ?